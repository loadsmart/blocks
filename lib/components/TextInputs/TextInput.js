"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importStar(require("react"));
var react_native_1 = require("react-native");
var res_1 = require("../../res/");
var TextInput = /** @class */ (function (_super) {
    __extends(TextInput, _super);
    function TextInput(props, state) {
        var _this = _super.call(this, props, state) || this;
        _this.state = {
            focused: false,
        };
        return _this;
    }
    TextInput.prototype.render = function () {
        var borderStyle = this.state.focused
            ? { borderColor: res_1.Colors.AlgaeGreen }
            : { borderColor: res_1.Colors.White };
        return (react_1.default.createElement(react_native_1.TextInput, __assign({}, this.props, { style: [styles.textInput, this.props.style, borderStyle], onFocus: this.onFocus.bind(this), onBlur: this.onBlur.bind(this) })));
    };
    TextInput.prototype.onFocus = function () {
        this.setState(function (_) { return ({
            focused: true,
        }); });
    };
    TextInput.prototype.onBlur = function () {
        this.setState(function (_) { return ({
            focused: false,
        }); });
    };
    return TextInput;
}(react_1.Component));
exports.default = TextInput;
var styles = react_native_1.StyleSheet.create({
    textInput: {
        height: 52,
        backgroundColor: res_1.Colors.White,
        borderRadius: 26,
        paddingHorizontal: 28,
        fontFamily: res_1.Fonts.SharpSansSemibold,
        fontSize: 16,
        color: res_1.Colors.Charcoal,
        borderWidth: 2,
    },
});
