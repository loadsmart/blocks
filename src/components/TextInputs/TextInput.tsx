import React, { Component } from 'react'
import { StyleSheet, TextInput as RNTextInput, TextInputProps } from 'react-native'
import { Colors, Fonts } from '../../res/'
import { ThemeContext } from '../Contexts'

interface Props extends TextInputProps {
  focusable?: boolean
}

interface State {
  focused: boolean
}

export default class TextInput extends Component<Props, State> {
  public static defaultProps: Props = {
    focusable: true,
  }

  constructor(props: Props, state: State) {
    super(props, state)
    this.state = {
      focused: false,
    }
  }

  public render() {
    return (
      <ThemeContext.Consumer>
        {theme => {
          const borderStyle =
            this.props.focusable === true && this.state.focused === true
              ? { borderColor: theme.primaryColor }
              : { borderColor: Colors.White }
          return (
            <RNTextInput
              {...this.props}
              style={[styles.textInput, this.props.style, borderStyle]}
              onFocus={this.onFocus.bind(this)}
              onBlur={this.onBlur.bind(this)}
            />
          )
        }}
      </ThemeContext.Consumer>
    )
  }

  public onFocus() {
    this.setState(_ => ({
      focused: true,
    }))
  }

  public onBlur() {
    this.setState(_ => ({
      focused: false,
    }))
  }
}

const styles = StyleSheet.create({
  textInput: {
    height: 52,
    backgroundColor: Colors.White,
    borderRadius: 26,
    paddingHorizontal: 28,
    fontFamily: Fonts.SharpSansSemibold,
    fontSize: 16,
    color: Colors.Charcoal,
    borderWidth: 2,
  },
})
