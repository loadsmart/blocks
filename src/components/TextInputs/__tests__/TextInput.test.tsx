import React from 'react'
import { TextInput as RNTextInput } from 'react-native'
import renderer from 'react-test-renderer'
import TextInput from '../TextInput'

describe('TextInput', () => {
  describe('when focusable is true', () => {
    it('renders with white border when on blur', () => {
      const testRenderer = renderer.create(<TextInput focusable />)
      const input = testRenderer.root.findByType(RNTextInput)
      input.props.onBlur()
      expect(testRenderer.toJSON()).toMatchSnapshot()
    })

    it('renders with green border when on focus', () => {
      const testRenderer = renderer.create(<TextInput focusable />)
      const input = testRenderer.root.findByType(RNTextInput)
      input.props.onFocus()
      expect(testRenderer.toJSON()).toMatchSnapshot()
    })
  })

  describe('when focusable is false', () => {
    it('renders with white border when on blur', () => {
      const testRenderer = renderer.create(<TextInput focusable={false} />)
      const input = testRenderer.root.findByType(RNTextInput)
      input.props.onBlur()
      expect(testRenderer.toJSON()).toMatchSnapshot()
    })

    it('also renders with white border when on focus', () => {
      const testRenderer = renderer.create(<TextInput focusable={false} />)
      const input = testRenderer.root.findByType(RNTextInput)
      input.props.onFocus()
      expect(testRenderer.toJSON()).toMatchSnapshot()
    })
  })
})
