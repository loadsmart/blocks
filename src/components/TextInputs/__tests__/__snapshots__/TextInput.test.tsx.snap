// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput when focusable is false also renders with white border when on focus 1`] = `
<TextInput
  allowFontScaling={true}
  focusable={false}
  onBlur={[Function]}
  onFocus={[Function]}
  rejectResponderTermination={true}
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 26,
        "borderWidth": 2,
        "color": "#3c3d3a",
        "fontFamily": "SharpSans-Semibold",
        "fontSize": 16,
        "height": 52,
        "paddingHorizontal": 28,
      },
      undefined,
      Object {
        "borderColor": "#fff",
      },
    ]
  }
  underlineColorAndroid="transparent"
/>
`;

exports[`TextInput when focusable is false renders with white border when on blur 1`] = `
<TextInput
  allowFontScaling={true}
  focusable={false}
  onBlur={[Function]}
  onFocus={[Function]}
  rejectResponderTermination={true}
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 26,
        "borderWidth": 2,
        "color": "#3c3d3a",
        "fontFamily": "SharpSans-Semibold",
        "fontSize": 16,
        "height": 52,
        "paddingHorizontal": 28,
      },
      undefined,
      Object {
        "borderColor": "#fff",
      },
    ]
  }
  underlineColorAndroid="transparent"
/>
`;

exports[`TextInput when focusable is true renders with green border when on focus 1`] = `
<TextInput
  allowFontScaling={true}
  focusable={true}
  onBlur={[Function]}
  onFocus={[Function]}
  rejectResponderTermination={true}
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 26,
        "borderWidth": 2,
        "color": "#3c3d3a",
        "fontFamily": "SharpSans-Semibold",
        "fontSize": 16,
        "height": 52,
        "paddingHorizontal": 28,
      },
      undefined,
      Object {
        "borderColor": "#14d64d",
      },
    ]
  }
  underlineColorAndroid="transparent"
/>
`;

exports[`TextInput when focusable is true renders with white border when on blur 1`] = `
<TextInput
  allowFontScaling={true}
  focusable={true}
  onBlur={[Function]}
  onFocus={[Function]}
  rejectResponderTermination={true}
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 26,
        "borderWidth": 2,
        "color": "#3c3d3a",
        "fontFamily": "SharpSans-Semibold",
        "fontSize": 16,
        "height": 52,
        "paddingHorizontal": 28,
      },
      undefined,
      Object {
        "borderColor": "#fff",
      },
    ]
  }
  underlineColorAndroid="transparent"
/>
`;
